Uninstallation.md

```diff
@@ -30,63 +30,94 @@ remove-->
-  # found if something goes wrong. In that case, look at the backed up
-  # files for PATH.
-  [System.Text.RegularExpressions.Regex]::Replace($userPath, [System.Text.RegularExpressions.Regex]::Escape("$env:ChocolateyInstall\bin") + '(?>;)?', '', [System.Text.RegularExpressions.RegexOptions]::IgnoreCase) | %{[System.Environment]::SetEnvironmentVariable('PATH', $_.Replace(";;",";"), 'User')}
+    Write-Verbose "Chocolatey Install location found in User Path. Removing..."
+    # WARNING: This could cause issues after reboot where nothing is
+    # found if something goes wrong. In that case, look at the backed up
+    # files for PATH.
+
+    $newUserPATH = ($userPath -split [System.IO.Path]::PathSeparator).Where{
+        $_ -ne "$env:ChocolateyInstall\bin"
+    } -join [System.IO.Path]::PathSeparator
+
+    [Environment]::SetEnvironmentVariable('PATH', $newUserPATH, 'User')
 }
 
 if ($machinePath -like "*$env:ChocolateyInstall*") {
-  Write-Output "Chocolatey Install location found in Machine Path. Removing..."
-  # WARNING: This could cause issues after reboot where nothing is
-  # found if something goes wrong. In that case, look at the backed up
-  # files for PATH.
-  [System.Text.RegularExpressions.Regex]::Replace($machinePath, [System.Text.RegularExpressions.Regex]::Escape("$env:ChocolateyInstall\bin") + '(?>;)?', '', [System.Text.RegularExpressions.RegexOptions]::IgnoreCase) | %{[System.Environment]::SetEnvironmentVariable('PATH', $_.Replace(";;",";"), 'Machine')}
+    Write-Verbose "Chocolatey Install location found in Machine Path. Removing..."
+    # WARNING: This could cause issues after reboot where nothing is
+    # found if something goes wrong. In that case, look at the backed up
+    # files for PATH.
+
+    $newMachinePATH = ($machinePath -split [System.IO.Path]::PathSeparator).Where{
+        -not [string]::IsNullOrEmpty($_) -and
+        $_ -ne "$env:ChocolateyInstall\bin"
+    } -join [System.IO.Path]::PathSeparator
+
+    [Environment]::SetEnvironmentVariable('PATH', $newMachinePATH, 'Machine')
 }
 
 # Adapt for any services running in subfolders of ChocolateyInstall
 $agentService = Get-Service -Name chocolatey-agent -ErrorAction SilentlyContinue
-if ($agentService -and $agentService.Status -eq 'Running') { $agentService.Stop() }
+if ($agentService -and $agentService.Status -eq 'Running') {
+    $agentService.Stop()
+}
 # TODO: add other services here
 
-# delete the contents (remove -WhatIf to actually remove)
 Remove-Item -Recurse -Force "$env:ChocolateyInstall" -WhatIf
 
-[System.Environment]::SetEnvironmentVariable("ChocolateyInstall", $null, 'User')
-[System.Environment]::SetEnvironmentVariable("ChocolateyInstall", $null, 'Machine')
-[System.Environment]::SetEnvironmentVariable("ChocolateyLastPathUpdate", $null, 'User')
-[System.Environment]::SetEnvironmentVariable("ChocolateyLastPathUpdate", $null, 'Machine')
+'ChocolateyInstall', 'ChocolateyLastPathUpdate', | ForEach-Object {
```

Comment:
Extra comma before the pipe?

I knew I spotted that befiore and forgot to mention it!

Reply:
Oof, good catch. That's what I get for doing touch ups in the online editor haha